builder/azure/arm/template_factory.go

```diff
@@ -51,6 +51,13 @@ func GetVirtualMachineDeployment(config *Config) (*resources.Deployment, error)
 		builder.SetMarketPlaceImage(config.ImagePublisher, config.ImageOffer, config.ImageSku, config.ImageVersion)
 	}
 
+	if config.VirtualNetworkName != "" {
+		builder.SetVirtualNetwork(
+			config.VirtualNetworkName,
```

Comment:
Huh.  I did try this out before I pushed the PR.  This error seems so flagrant (read embarrassing).  I did let the code find the appropriate Resource Group and Subnet as opposed to setting it myself.

I have not retried since updating the code based on Paul's comments.  

Thank you for trying this.  I'll take a look at the PR.